pkg/auth/providers/publicapi/login.go

```diff
@@ -208,9 +213,61 @@ func (h *loginHandler) createLoginToken(request *types.APIContext) (v3.Token, st
 		if err != nil {
 			return v3.Token{}, "", "", err
 		}
+
+		if err := h.createClusterAuthTokenIfNeeded(token, tokenValue); err != nil {
+			logrus.Infof("Failed to create cluster auth token for cluster [%s], cluster auth endpoint may fail: %v", token.ClusterName, err)
+		}
 		return *token, tokenValue, responseType, nil
 	}
 
 	rToken, unhashedTokenKey, err := h.tokenMGR.NewLoginToken(currUser.Name, userPrincipal, groupPrincipals, providerToken, ttl, description)
 	return rToken, unhashedTokenKey, responseType, err
 }
+
+// createClusterAuthTokenIfNeeded checks if local cluster auth endpoint is enabled. If it is, a cluster auth token
+// is enabled.
```

Comment:
```suggestion
// is created.
```

Reply:
? auth endpoint is a bool on the cluster object, I think enabled makes sense.

Reply:
If the _endpoint_ is enabled it _creates_ a token though right ? I don't understand why you would say the token was enabled

Reply:
Ohhh I see, I read the first sentence with "enabled". Good catch.